### ISSUE TITLE
Doesn't detect that `.flatMap` is not supported 

### ISSUE THREAD

**Issue by anton164** [amilajack/eslint-plugin-compat#150]
I guess this is related to #72, but without this the plugin is pretty much useless for our usecase, because it's not able to detect browser compatibility..

**Code snippet.**
```
  getTargetComponenets() {
    return this._getTargetWorkpsaces()
      .filter(workspace => !this._disabledWorkspaces.has(workspace.id))
      .flatMap(workspace => Components.collection.getWorkspaceComponents(null, workspace));
  }
```

`.flatMap` support: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap

**Comment by vikr01** [comment]
See [#93 (comment)](https://github.com/amilajack/eslint-plugin-compat/issues/93#issuecomment-318669186). To know `.flatMap` exists, the AST would need to know the return type of `this._getTargetWorkspaces` and `.filter` of that return value. This isn't easy to implement with `eslint`.  Integrating a type system to do something like this is a possible solution and has already been considered ([#70](https://github.com/amilajack/eslint-plugin-compat/issues/70#issuecomment-286644988)).

> the plugin is pretty much useless for our usecase

Then don't use it.
